core/server/master/src/main/java/alluxio/master/metastore/BlockStore.java

```diff
@@ -87,6 +87,27 @@
    */
   long size();
 
+  /**
+   * Gets a {@link CloseableIterator} over the blocks.
+   * The iterator must be closed properly.
+   * One option is to follow the below idiom:
+   * <pre>{@code
+   *   try (CloseableIterator<Block> iter =
+   *       mBlockStore.getCloseableIterator()) {
+   *     for (; iter.hasNext(); ) {
+   *       // take the element and perform operations
+   *     }
+   *   }
+   * }</pre>
+   *
+   * If the iterator must be passed to other methods,
+   * it must be closed at the end of operation or on exceptions.
+   * Otherwise there can be a leak!
+   *
+   * @return a {@link CloseableIterator} over the blocks
+   * */
+  CloseableIterator<Block> getCloseableIterator();
+
```

Comment:
I noticed this interface has a `close()` method. Is there/should there be any coupling of the lifecycle of `BlockStore` and the lifecycle of `getCloseableIterator()`?

Reply:
I'd guess no, now that iterators that are exported are closeable as well..

Reply:
Right, the iterators given out from `getCloseableIterator()` (before this change the iterators were given out by `iterator()` ) have independent life cycles. The `BlockStore` does not track and close those. 

In other words, the iterators should be closed as soon as they are done, instead of waiting for the `BlockStore.close()` to close those.